x-pack/plugins/canvas/canvas_plugin_src/functions/common/joinRows.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { ExpressionFunction } from '../../../../../../src/legacy/core_plugins/interpreter/types';
+import { Datatable } from '../types';
+import { getFunctionErrors } from '../../strings/functions';
+
+interface Arguments {
+  column: string;
+  separator: string;
+  quote: string;
+}
+
+const escapeString = (data: string, quotechar: string): string => {
+  if (quotechar === undefined || quotechar === '') {
+    return data;
+  } else {
+    const regex = new RegExp(quotechar, 'g');
+    return data.replace(/\\/g, `\\\\`).replace(regex, `\\${quotechar}`);
+  }
+};
+
+export function joinRows(): ExpressionFunction<'joinRows', Datatable, Arguments, string> {
+  const errors = getFunctionErrors().getCell;
+  return {
+    name: 'joinRows',
+    type: 'string',
+    help: 'Join values from rows in a datatable into a string',
+    context: {
+      types: ['datatable'],
+    },
+    args: {
+      column: {
+        aliases: ['_'],
+        types: ['string'],
+        required: true,
+        help: 'Column to join values from',
+      },
+      separator: {
+        aliases: ['sep'],
+        types: ['string'],
+        default: ',',
+        help: 'Separator to use for row values',
+      },
+      quote: {
+        types: ['string'],
+        default: 'DEFAULT',
+        help: 'Quote character around values',
+      },
+    },
+    fn: (context, { column, separator, quote }) => {
+      // work around having single quote as the default value not working.
+      const quotechar = quote === 'DEFAULT' ? "'" : quote;
```

Comment:
Hit an issue when I tried to set the default value for quote to `"'"` because it tries to parse it as a canvas expression.

`Unable to parse expression: Expected \"$\", \"{\", or literal but \"`
in `interpret.js`
`argAsts[argName] = [fromExpression(argDef.default, 'argument')];`

Reply:
That's an interesting way to handle that, but breaks from convention a bit. In some of the existing functions, we simply don't define a default and assign it in code if the value `== null`. You can see this in the `csv` function. I think that's probably better, otherwise when we generate docs, or users look at the auto-complete info, it'll say the default is the string "DEFAULT", which isn't really true. 

I haven't tried it yet, but I think using `"\""` might also work, but may suffer from the same misinformation problem.

@cqliu1 @monfera what do you two think?

Reply:
Can it be made to work, following the example for the convention @w33ble mentioned?

Reply:
This was just a workaround because the interpreter didn't like my quote, but @cqliu1's solution with quoting worked.